# <p style="padding:10px;background-color:#0fa79f;margin:0;color:#111423;font-family:newtimeroman;font-size:150%;text-align:center;border-radius: 15px 50px;overflow:hidden;font-weight:500">InsiderStore CASE</p>

In [2]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
import matplotlib

In [3]:
print(f'Pandas version: {pd.__version__}')
print(f'Numpy version: {np.__version__}')
print(f'Seaborn version: {sns.__version__}')
print(f'Matplotlib version: {matplotlib.__version__}')

Pandas version: 1.4.4
Numpy version: 1.21.5
Seaborn version: 0.11.2
Matplotlib version: 3.5.2


<a id="3"></a>
# <p style="padding:10px;background-color:#0fa79f;margin:0;color:#111423;font-family:newtimeroman;font-size:100%;text-align:center;border-radius: 15px 50px;overflow:hidden;font-weight:500">Initial Data Inspection</p>

**Importing the datasets**

<center> <span style="font-family:newtimeroman"> <span style="padding:5px;color:white;display:fill;border-radius:20px 50px;background-color:#FDD667;font-size:200%;font-weight: 500;color:#111423;overflow:hidden;">ㅤClient Base Datasetㅤ</span>

In [15]:
clients = pd.read_csv("data/Base Case CRM 2022 - base_clientes.csv")

In [21]:
clients.head()

Email Sexo          Cidade Estado
0  cliente71@yahoo.com.br    M       São Paulo     SP
1  cliente129@outlook.com    M        Londrina     PR
2   cliente150@bol.com.br    F  Rio de Janeiro     RJ
3   cliente231@bol.com.br    M          Recife     PE
4   cliente12@outlook.com    F   Caraguatatuba     SP

In [18]:
clients.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 297 entries, 0 to 296
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Email   297 non-null    object
 1   Sexo    274 non-null    object
 2   Cidade  297 non-null    object
 3   Estado  297 non-null    object
dtypes: object(4)
memory usage: 9.4+ KB


In [48]:
clients['Sexo'].value_counts()

M    189
F     85
Name: Sexo, dtype: int64

In [56]:
clients['Estado'].value_counts()

SP    117
RJ     30
MG     27
PR     24
AM     24
CE     24
PE     21
SC     12
RN      9
DF      6
GO      3
Name: Estado, dtype: int64

In [62]:
clients['Cidade'].value_counts()

São Paulo              48
Fortaleza              24
Manaus                 24
Recife                 21
São José dos Campos    21
Cotia                  18
Rio de Janeiro         18
Campinas               15
Curitiba               15
Belo Horizonte         15
Florianópolis          12
Angra dos Reis         12
Caraguatatuba          12
Londrina                9
Contagem                9
Natal                   9
Brasília                6
Jacareí                 3
Goiânia                 3
Cordisburgo             3
Name: Cidade, dtype: int64

**Insights:**
- 297 unique clients, 189 male (63.6%), 85 female (28.6%) and 23 undeclared (7.7%)
- 40% of all clients from SP, followed by ~10% each from RJ, MG. Overall, the SE region accounts for ~60% of clients.

**Additional checks below**

In [63]:
clients.isna().sum()

Email      0
Sexo      23
Cidade     0
Estado     0
dtype: int64

**Final Remarks**
- No cleaning necessary. NaNs in Sexo column indicative of clients who prefer not to declare binary genders.

<center> <span style="font-family:newtimeroman"> <span style="padding:5px;color:white;display:fill;border-radius:20px 50px;background-color:#FDD667;font-size:200%;font-weight: 500;color:#111423;overflow:hidden;">ㅤItem Base Datasetㅤ</span>

In [31]:
items = pd.read_excel("data/Base Case CRM 2022 - base_itens.xlsx")

In [22]:
items.head()

Order ID Cupom             Order_email Data Compra  SKU  qtd  \
0  IN-1288   NaN    cliente141@gmail.com  2021-10-01   11    1   
1  IN-1341   NaN  cliente212@outlook.com  2022-02-08   10    1   
2  IN-1438   NaN    cliente251@gmail.com  2022-02-03    8    3   
3  IN-1218   NaN   cliente97@outlook.com  2021-10-20   11    3   
4  IN-1207   NaN  cliente211@outlook.com  2022-02-04   13    1   

   preco_unitario  subtotal  desconto  total  
0              60        60       0.0   60.0  
1             145       145       0.0  145.0  
2             130       390       0.0  390.0  
3              60       180       0.0  180.0  
4              70        70       0.0   70.0

In [20]:
items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 802 entries, 0 to 801
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Order ID        802 non-null    object        
 1   Cupom           113 non-null    object        
 2   Order_email     802 non-null    object        
 3   Data Compra     802 non-null    datetime64[ns]
 4   SKU             802 non-null    int64         
 5   qtd             802 non-null    int64         
 6   preco_unitario  802 non-null    int64         
 7   subtotal        802 non-null    int64         
 8   desconto        802 non-null    float64       
 9   total           802 non-null    float64       
dtypes: datetime64[ns](1), float64(2), int64(4), object(3)
memory usage: 62.8+ KB


In [68]:
items['Order ID'].value_counts()

IN-1263    4
IN-1030    4
IN-1063    3
IN-1197    3
IN-1094    3
          ..
IN-1119    1
IN-1117    1
IN-1351    1
IN-1343    1
IN-1274    1
Name: Order ID, Length: 550, dtype: int64

In [69]:
items['Order_email'].value_counts()

cliente251@gmail.com       17
cliente289@gmail.com       16
cliente141@gmail.com       15
cliente68@gmail.com        15
cliente195@yahoo.com.br    15
                           ..
cliente105@bol.com.br       1
cliente161@yahoo.com.br     1
cliente82@outlook.com       1
cliente226@yahoo.com.br     1
cliente295@yahoo.com.br     1
Name: Order_email, Length: 272, dtype: int64

**Insights:**
- 15% of orders made with Coupons
- There are repeated Order ID's. Implying..?
- Some clients ordered only once, others ordered items over 15 times each. Keep those in mind.

**Additional checks below**

In [65]:
items.isna().sum()

Order ID            0
Cupom             689
Order_email         0
Data Compra         0
SKU                 0
qtd                 0
preco_unitario      0
subtotal            0
desconto            0
total               0
dtype: int64

**Final Remarks**
- No cleaning necessary.
- Create histograms over date to check for pattens.
- Analyze total revenue and impact of discounts, as well as any relationships with clients.

<center> <span style="font-family:newtimeroman"> <span style="padding:5px;color:white;display:fill;border-radius:20px 50px;background-color:#FDD667;font-size:200%;font-weight: 500;color:#111423;overflow:hidden;">ㅤDim_product Datasetㅤ</span>

In [23]:
dim = pd.read_excel("data/Base Case CRM 2022 - dim_product.xlsx")

In [70]:
dim

SKU              Nome do Produto     Gênero   Categoria  Preço
0     1  T-Shirt Esportiva Masculina  Masculino  Esportivos    160
1     2               Motions Shorts   Feminino  Esportivos    210
2     3                      Legging   Feminino  Esportivos    300
3     4                          Top   Feminino  Esportivos    200
4     5            Cueca Performance  Masculino  Esportivos     70
5     6                         Meia        NaN      Casual     45
6     7                     Wingsuit   Feminino      Casual    400
7     8            T-Shirt Masculina  Masculino      Casual    130
8     9             T-Shirt Feminina   Feminino      Casual    135
9    10          T-Shirt Manga Longa   Feminino      Casual    145
10   11                Cueca Simples  Masculino   Underwear     60
11   12                   Sutiã Flex   Feminino   Underwear    150
12   13                     Calcinha   Feminino   Underwear     75

In [25]:
dim.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13 entries, 0 to 12
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   SKU              13 non-null     int64 
 1   Nome do Produto  13 non-null     object
 2   Gênero           12 non-null     object
 3   Categoria        13 non-null     object
 4   Preço            13 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 648.0+ bytes


In [72]:
dim['Gênero'].value_counts()

Feminino     8
Masculino    4
Name: Gênero, dtype: int64

In [73]:
dim['Categoria'].value_counts()

Esportivos    5
Casual        5
Underwear     3
Name: Categoria, dtype: int64

**Insights:**
- 13 Insider unique products
- 8 for women, 4 for men, 1 neutral (socks)
- Types: 5 sports, 5 casual, 3 underwear

**Final Remarks**
- No cleaning necessary.
- Cross with sales to find most requested items.

<center> <span style="font-family:newtimeroman"> <span style="padding:5px;color:white;display:fill;border-radius:20px 50px;background-color:#FDD667;font-size:200%;font-weight: 500;color:#111423;overflow:hidden;">ㅤFebruary Datasetㅤ</span>

In [83]:
fev = pd.read_excel("data/Case CRM Fev 2022.xlsx")

In [29]:
fev.head()

Unnamed: 0                               Unnamed: 1  Unnamed: 2  \
0         NaN                                      NaN         NaN   
1         NaN                           Tema/Categoria  Data envio   
2         NaN                  Camiseta H - last hours  [10022022]   
3         NaN                Camiseta H - promo 10%off  [10022022]   
4         NaN  Back in stock H / Lancamento camiseta M  [07022022]   

     Unnamed: 3       Unnamed: 4  Unnamed: 5        Unnamed: 6    Unnamed: 7  
0           NaN              NaN         NaN               NaN           NaN  
1      # envios  # unsubscribers   # cliques  Taxa de abertura       Receita  
2  46964.395537        46.791553  593.011318          0.230711  35571.702739  
3  30283.684262        33.422538  482.239478          0.200058  25001.930123  
4  48276.468888        69.709865  741.980345          0.271296  23923.193834

In [30]:
fev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  0 non-null      float64
 1   Unnamed: 1  47 non-null     object 
 2   Unnamed: 2  46 non-null     object 
 3   Unnamed: 3  47 non-null     object 
 4   Unnamed: 4  47 non-null     object 
 5   Unnamed: 5  47 non-null     object 
 6   Unnamed: 6  47 non-null     object 
 7   Unnamed: 7  47 non-null     object 
dtypes: float64(1), object(7)
memory usage: 3.1+ KB


In [84]:
fev.drop(labels='Unnamed: 0',axis=1,inplace=True)

In [86]:
fev.drop(labels=0,axis=0,inplace=True)

In [87]:
fev.head()

Unnamed: 1  Unnamed: 2    Unnamed: 3  \
1                           Tema/Categoria  Data envio      # envios   
2                  Camiseta H - last hours  [10022022]  46964.395537   
3                Camiseta H - promo 10%off  [10022022]  30283.684262   
4  Back in stock H / Lancamento camiseta M  [07022022]  48276.468888   
5                      Cor nova camiseta H  [07022022]  47609.927986   

        Unnamed: 4  Unnamed: 5        Unnamed: 6    Unnamed: 7  
1  # unsubscribers   # cliques  Taxa de abertura       Receita  
2        46.791553  593.011318          0.230711  35571.702739  
3        33.422538  482.239478          0.200058  25001.930123  
4        69.709865  741.980345          0.271296  23923.193834  
5        77.349302  737.205696          0.253438  40891.997838

In [91]:
fev = fev.rename(columns=fev.iloc[0]).drop(fev.index[0])

In [92]:
fev.head()

Tema/Categoria  Data envio      # envios  \
2                  Camiseta H - last hours  [10022022]  46964.395537   
3                Camiseta H - promo 10%off  [10022022]  30283.684262   
4  Back in stock H / Lancamento camiseta M  [07022022]  48276.468888   
5                      Cor nova camiseta H  [07022022]  47609.927986   
6             Recap Semanal Camiseta ambos  [13022022]  34620.974771   

  # unsubscribers   # cliques Taxa de abertura       Receita  
2       46.791553  593.011318         0.230711  35571.702739  
3       33.422538  482.239478         0.200058  25001.930123  
4       69.709865  741.980345         0.271296  23923.193834  
5       77.349302  737.205696         0.253438  40891.997838  
6       65.890146  486.059196         0.238159   21179.75732

In [93]:
fev.columns

Index(['Tema/Categoria', 'Data envio', '# envios', '# unsubscribers',
       '# cliques', 'Taxa de abertura', 'Receita'],
      dtype='object')

In [104]:
fev['Data envio'] = fev['Data envio'].str.lstrip('[')

In [105]:
fev['Data envio'] = fev['Data envio'].str.rstrip(']')

In [112]:
fev['Data envio'] = pd.to_datetime(fev['Data envio'],format='%d%m%Y')

In [115]:
fev.head() #date converted to datetime

Tema/Categoria Data envio      # envios  \
2                  Camiseta H - last hours 2022-02-10  46964.395537   
3                Camiseta H - promo 10%off 2022-02-10  30283.684262   
4  Back in stock H / Lancamento camiseta M 2022-02-07  48276.468888   
5                      Cor nova camiseta H 2022-02-07  47609.927986   
6             Recap Semanal Camiseta ambos 2022-02-13  34620.974771   

  # unsubscribers   # cliques Taxa de abertura       Receita  
2       46.791553  593.011318         0.230711  35571.702739  
3       33.422538  482.239478         0.200058  25001.930123  
4       69.709865  741.980345         0.271296  23923.193834  
5       77.349302  737.205696         0.253438  40891.997838  
6       65.890146  486.059196         0.238159   21179.75732

In [114]:
fev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 46 entries, 2 to 47
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Tema/Categoria    46 non-null     object        
 1   Data envio        45 non-null     datetime64[ns]
 2   # envios          46 non-null     object        
 3   # unsubscribers   46 non-null     object        
 4   # cliques         46 non-null     object        
 5   Taxa de abertura  46 non-null     object        
 6   Receita           46 non-null     object        
dtypes: datetime64[ns](1), object(6)
memory usage: 2.6+ KB


In [117]:
fev['# envios'] = fev['# envios'].astype(int)

In [118]:
fev['# unsubscribers'] = fev['# unsubscribers'].astype(int)

In [119]:
fev['# cliques'] = fev['# cliques'].astype(int)


In [123]:
fev['Taxa de abertura'] = fev['Taxa de abertura'].astype(float).round(2)

**Insights:**
- 297 unique clients, 189 male (63.6%), 85 female (28.6%) and 23 undeclared (7.7%)
- 40% of all clients from SP, followed by ~10% each from RJ, MG. Overall, the SE region accounts for ~60% of clients.

**Final Remarks**
- Some cleaning required. New header, dropped rows/columns of NaNs.
- Date column converted to datetime
- Rounded numerical columns 
- Exported as new clean xlsx

In [95]:
fev_clean = fev

In [ ]:
fev_clean.to_csv('fev_clean.csv')